datafusion-cli/Dockerfile

```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-FROM rust:1.73-bullseye as builder
+FROM rust:1.78-bookworm as builder
```

Comment:
The workspace.package in the `Cargo.toml` is still 1.73.  https://github.com/apache/datafusion/blob/main/Cargo.toml#L53

I am fine with updating the Debian version, but updating the Rust version might require more thoughts or having more extensive implications cc @alamb @andygrove

Reply:
The Rust version update is only for the build of the CLI. Nothing more. It's not related to the MSRV of DataFusion unless the build of the CLI is the litmus test that the MSRV we have actually builds and runs.

Reply:
> The Rust version update is only for the build of the CLI. Nothing more. It's not related to the MSRV of DataFusion unless the build of the CLI is the litmus test that the MSRV we have actually builds and runs.

I think the MSRV is verified via a CI job:  https://github.com/apache/datafusion/blob/4b7b5ab11a868c0069245f411f0d9b08e1b2744e/.github/workflows/rust.yml#L556-L584